code/modules/urist/modules/shipbattles/ship_overmaps.dm

```diff
@@ -17,6 +17,8 @@
 	var/can_escape = TRUE
 	var/dam_announced = 0
 	var/pvp_cooldown = 0
+	var/list/target_dirs //what directions can we be shot from. this is to mimic the effect of the old landmark system that mostly prevented hits on the SM.  if this is empty it will use GLOB.cardinal
+	var/list/target_zs //what zs can be hit in ship combat. this is mainly here to prevent the top deck getting hit on nerva.
```

Comment:
You're right that that might be a better way to handle this, but that's probably best left for a future PR that expands on PVP, because that method would take a lot more work to function with how awaymap zs are loaded. This is fine for now because it mimics the existing landmark layout, where there's no landmarks at the rear anyways because of z-level explosions,